File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/view_permission_summary.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class ViewPermissionSummary(object):
    """
    Information about a view permission.
    """

    #: A constant which can be used with the grantee_type property of a ViewPermissionSummary.
    #: This constant has a value of "USER"
    GRANTEE_TYPE_USER = "USER"

    #: A constant which can be used with the grantee_type property of a ViewPermissionSummary.
    #: This constant has a value of "ROLE"
    GRANTEE_TYPE_ROLE = "ROLE"

    #: A constant which can be used with the grantee_type property of a ViewPermissionSummary.
    #: This constant has a value of "GROUP"
    GRANTEE_TYPE_GROUP = "GROUP"

    #: A constant which can be used with the grantee_permissions property of a ViewPermissionSummary.
    #: This constant has a value of "SELECT"
    GRANTEE_PERMISSIONS_SELECT = "SELECT"

    #: A constant which can be used with the grantee_permissions property of a ViewPermissionSummary.
    #: This constant has a value of "MANAGE"
    GRANTEE_PERMISSIONS_MANAGE = "MANAGE"

    #: A constant which can be used with the grantee_permissions property of a ViewPermissionSummary.
    #: This constant has a value of "WRITE"
    GRANTEE_PERMISSIONS_WRITE = "WRITE"

    #: A constant which can be used with the grantee_permissions property of a ViewPermissionSummary.
    #: This constant has a value of "INSERT"
    GRANTEE_PERMISSIONS_INSERT = "INSERT"

    #: A constant which can be used with the grantee_permissions property of a ViewPermissionSummary.
    #: This constant has a value of "UPDATE"
    GRANTEE_PERMISSIONS_UPDATE = "UPDATE"

    #: A constant which can be used with the grantee_permissions property of a ViewPermissionSummary.
    #: This constant has a value of "DELETE"
    GRANTEE_PERMISSIONS_DELETE = "DELETE"

    #: A constant which can be used with the grantee_permissions property of a ViewPermissionSummary.
    #: This constant has a value of "ALTER"
    GRANTEE_PERMISSIONS_ALTER = "ALTER"

    #: A constant which can be used with the grantee_permissions property of a ViewPermissionSummary.
    #: This constant has a value of "ADMIN"
    GRANTEE_PERMISSIONS_ADMIN = "ADMIN"

    def __init__(self, **kwargs):
        """
        Initializes a new ViewPermissionSummary object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param grantee:
            The value to assign to the grantee property of this ViewPermissionSummary.
        :type grantee: str

        :param grantee_name:
            The value to assign to the grantee_name property of this ViewPermissionSummary.
        :type grantee_name: str

        :param grantee_type:
            The value to assign to the grantee_type property of this ViewPermissionSummary.
            Allowed values for this property are: "USER", "ROLE", "GROUP", 'UNKNOWN_ENUM_VALUE'.
            Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.
        :type grantee_type: str

        :param grantee_permissions:
            The value to assign to the grantee_permissions property of this ViewPermissionSummary.
            Allowed values for items in this list are: "SELECT", "MANAGE", "WRITE", "INSERT", "UPDATE", "DELETE", "ALTER", "ADMIN", 'UNKNOWN_ENUM_VALUE'.
            Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.
        :type grantee_permissions: list[str]

        :param columns:
            The value to assign to the columns property of this ViewPermissionSummary.
        :type columns: list[str]

        :param excluded_columns:
            The value to assign to the excluded_columns property of this ViewPermissionSummary.
        :type excluded_columns: list[str]

        :param is_inherited:
            The value to assign to the is_inherited property of this ViewPermissionSummary.
        :type is_inherited: bool

        :param resource_name:
            The value to assign to the resource_name property of this ViewPermissionSummary.
        :type resource_name: str

        """
        self.swagger_types = {
            'grantee': 'str',
            'grantee_name': 'str',
            'grantee_type': 'str',
            'grantee_permissions': 'list[str]',
            'columns': 'list[str]',
            'excluded_columns': 'list[str]',
            'is_inherited': 'bool',
            'resource_name': 'str'
        }

        self.attribute_map = {
            'grantee': 'grantee',
            'grantee_name': 'granteeName',
            'grantee_type': 'granteeType',
            'grantee_permissions': 'granteePermissions',
            'columns': 'columns',
            'excluded_columns': 'excludedColumns',
            'is_inherited': 'isInherited',
            'resource_name': 'resourceName'
        }

        self._grantee = None
        self._grantee_name = None
        self._grantee_type = None
        self._grantee_permissions = None
        self._columns = None
        self._excluded_columns = None
        self._is_inherited = None
        self._resource_name = None

    @property
    def grantee(self):
        """
        **[Required]** Gets the grantee of this ViewPermissionSummary.
        The OCID of user/group and name in case of role.


        :return: The grantee of this ViewPermissionSummary.
        :rtype: str
        """
        return self._grantee

    @grantee.setter
    def grantee(self, grantee):
        """
        Sets the grantee of this ViewPermissionSummary.
        The OCID of user/group and name in case of role.


        :param grantee: The grantee of this ViewPermissionSummary.
        :type: str
        """
        self._grantee = grantee

    @property
    def grantee_name(self):
        """
        Gets the grantee_name of this ViewPermissionSummary.
        The simplified name of the grantee.


        :return: The grantee_name of this ViewPermissionSummary.
        :rtype: str
        """
        return self._grantee_name

    @grantee_name.setter
    def grantee_name(self, grantee_name):
        """
        Sets the grantee_name of this ViewPermissionSummary.
        The simplified name of the grantee.


        :param grantee_name: The grantee_name of this ViewPermissionSummary.
        :type: str
        """
        self._grantee_name = grantee_name

    @property
    def grantee_type(self):
        """
        **[Required]** Gets the grantee_type of this ViewPermissionSummary.
        The type of grantee.

        Allowed values for this property are: "USER", "ROLE", "GROUP", 'UNKNOWN_ENUM_VALUE'.
        Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.


        :return: The grantee_type of this ViewPermissionSummary.
        :rtype: str
        """
        return self._grantee_type

    @grantee_type.setter
    def grantee_type(self, grantee_type):
        """
        Sets the grantee_type of this ViewPermissionSummary.
        The type of grantee.


        :param grantee_type: The grantee_type of this ViewPermissionSummary.
        :type: str
        """
        allowed_values = ["USER", "ROLE", "GROUP"]
        if not value_allowed_none_or_none_sentinel(grantee_type, allowed_values):
            grantee_type = 'UNKNOWN_ENUM_VALUE'
        self._grantee_type = grantee_type

    @property
    def grantee_permissions(self):
        """
        **[Required]** Gets the grantee_permissions of this ViewPermissionSummary.
        The selected permissions for a view.

        Allowed values for items in this list are: "SELECT", "MANAGE", "WRITE", "INSERT", "UPDATE", "DELETE", "ALTER", "ADMIN", 'UNKNOWN_ENUM_VALUE'.
        Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.


        :return: The grantee_permissions of this ViewPermissionSummary.
        :rtype: list[str]
        """
        return self._grantee_permissions

    @grantee_permissions.setter
    def grantee_permissions(self, grantee_permissions):
        """
        Sets the grantee_permissions of this ViewPermissionSummary.
        The selected permissions for a view.


        :param grantee_permissions: The grantee_permissions of this ViewPermissionSummary.
        :type: list[str]
        """
        allowed_values = ["SELECT", "MANAGE", "WRITE", "INSERT", "UPDATE", "DELETE", "ALTER", "ADMIN"]
        if grantee_permissions:
            grantee_permissions[:] = ['UNKNOWN_ENUM_VALUE' if not value_allowed_none_or_none_sentinel(x, allowed_values) else x for x in grantee_permissions]
        self._grantee_permissions = grantee_permissions

    @property
    def columns(self):
        """
        Gets the columns of this ViewPermissionSummary.
        The list of the columns included for permission assignment.


        :return: The columns of this ViewPermissionSummary.
        :rtype: list[str]
        """
        return self._columns

    @columns.setter
    def columns(self, columns):
        """
        Sets the columns of this ViewPermissionSummary.
        The list of the columns included for permission assignment.


        :param columns: The columns of this ViewPermissionSummary.
        :type: list[str]
        """
        self._columns = columns

    @property
    def excluded_columns(self):
        """
        Gets the excluded_columns of this ViewPermissionSummary.
        The list of the columns excluded from permission assignment.


        :return: The excluded_columns of this ViewPermissionSummary.
        :rtype: list[str]
        """
        return self._excluded_columns

    @excluded_columns.setter
    def excluded_columns(self, excluded_columns):
        """
        Sets the excluded_columns of this ViewPermissionSummary.
        The list of the columns excluded from permission assignment.


        :param excluded_columns: The excluded_columns of this ViewPermissionSummary.
        :type: list[str]
        """
        self._excluded_columns = excluded_columns

    @property
    def is_inherited(self):
        """
        Gets the is_inherited of this ViewPermissionSummary.
        The permission listed is inherited or not from object up in hierarchy.


        :return: The is_inherited of this ViewPermissionSummary.
        :rtype: bool
        """
        return self._is_inherited

    @is_inherited.setter
    def is_inherited(self, is_inherited):
        """
        Sets the is_inherited of this ViewPermissionSummary.
        The permission listed is inherited or not from object up in hierarchy.


        :param is_inherited: The is_inherited of this ViewPermissionSummary.
        :type: bool
        """
        self._is_inherited = is_inherited

    @property
    def resource_name(self):
        """
        Gets the resource_name of this ViewPermissionSummary.
        Name of the object to which this permission belong to. This would be the name of view if permission is not inherited or name of object up in hierarchy if permission is inherited.


        :return: The resource_name of this ViewPermissionSummary.
        :rtype: str
        """
        return self._resource_name

    @resource_name.setter
    def resource_name(self, resource_name):
        """
        Sets the resource_name of this ViewPermissionSummary.
        Name of the object to which this permission belong to. This would be the name of view if permission is not inherited or name of object up in hierarchy if permission is inherited.


        :param resource_name: The resource_name of this ViewPermissionSummary.
        :type: str
        """
        self._resource_name = resource_name

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/ViewPermissionSummary.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Information about a view permission.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=ViewPermissionSummary.Builder.class)

public final class ViewPermissionSummary  {
    @Deprecated
    @java.beans.ConstructorProperties({"grantee", "granteeName", "granteeType", "granteePermissions", "columns", "excludedColumns", "isInherited", "resourceName"})
    public ViewPermissionSummary(String grantee, String granteeName, GranteeType granteeType, java.util.List<ViewPrivilege> granteePermissions, java.util.List<String> columns, java.util.List<String> excludedColumns, Boolean isInherited, String resourceName) {
        super();
        this.grantee = grantee;
        this.granteeName = granteeName;
        this.granteeType = granteeType;
        this.granteePermissions = granteePermissions;
        this.columns = columns;
        this.excludedColumns = excludedColumns;
        this.isInherited = isInherited;
        this.resourceName = resourceName;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * The OCID of user/group and name in case of role.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("grantee")
        private String grantee;

                /**
         * The OCID of user/group and name in case of role.
         * @param grantee the value to set
         * @return this builder
         **/
        

        public Builder grantee(String grantee) {
        this.grantee = grantee;
        return this;
        }
            /**
     * The simplified name of the grantee.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("granteeName")
        private String granteeName;

                /**
         * The simplified name of the grantee.
         * @param granteeName the value to set
         * @return this builder
         **/
        

        public Builder granteeName(String granteeName) {
        this.granteeName = granteeName;
        return this;
        }
            /**
     * The type of grantee.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("granteeType")
        private GranteeType granteeType;

                /**
         * The type of grantee.
         * @param granteeType the value to set
         * @return this builder
         **/
        

        public Builder granteeType(GranteeType granteeType) {
        this.granteeType = granteeType;
        return this;
        }
            /**
     * The selected permissions for a view.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("granteePermissions")
        private java.util.List<ViewPrivilege> granteePermissions;

                /**
         * The selected permissions for a view.
         * @param granteePermissions the value to set
         * @return this builder
         **/
        

        public Builder granteePermissions(java.util.List<ViewPrivilege> granteePermissions) {
        this.granteePermissions = granteePermissions;
        return this;
        }
            /**
     * The list of the columns included for permission assignment.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("columns")
        private java.util.List<String> columns;

                /**
         * The list of the columns included for permission assignment.
         * @param columns the value to set
         * @return this builder
         **/
        

        public Builder columns(java.util.List<String> columns) {
        this.columns = columns;
        return this;
        }
            /**
     * The list of the columns excluded from permission assignment.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("excludedColumns")
        private java.util.List<String> excludedColumns;

                /**
         * The list of the columns excluded from permission assignment.
         * @param excludedColumns the value to set
         * @return this builder
         **/
        

        public Builder excludedColumns(java.util.List<String> excludedColumns) {
        this.excludedColumns = excludedColumns;
        return this;
        }
            /**
     * The permission listed is inherited or not from object up in hierarchy.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("isInherited")
        private Boolean isInherited;

                /**
         * The permission listed is inherited or not from object up in hierarchy.
         * @param isInherited the value to set
         * @return this builder
         **/
        

        public Builder isInherited(Boolean isInherited) {
        this.isInherited = isInherited;
        return this;
        }
            /**
     * Name of the object to which this permission belong to. This would be the name of view if permission is not inherited or name of object up in hierarchy if permission is inherited.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("resourceName")
        private String resourceName;

                /**
         * Name of the object to which this permission belong to. This would be the name of view if permission is not inherited or name of object up in hierarchy if permission is inherited.
         * @param resourceName the value to set
         * @return this builder
         **/
        

        public Builder resourceName(String resourceName) {
        this.resourceName = resourceName;
        return this;
        }


        public ViewPermissionSummary build() {
            ViewPermissionSummary model = new ViewPermissionSummary(this.grantee
                    , this.granteeName
                    , this.granteeType
                    , this.granteePermissions
                    , this.columns
                    , this.excludedColumns
                    , this.isInherited
                    , this.resourceName);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(ViewPermissionSummary model) {
                this.grantee(model.getGrantee());
    this.granteeName(model.getGranteeName());
    this.granteeType(model.getGranteeType());
    this.granteePermissions(model.getGranteePermissions());
    this.columns(model.getColumns());
    this.excludedColumns(model.getExcludedColumns());
    this.isInherited(model.getIsInherited());
    this.resourceName(model.getResourceName());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The OCID of user/group and name in case of role.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("grantee")
    private final String grantee;

        /**
     * The OCID of user/group and name in case of role.
     * @return the value
     **/
    
    public String getGrantee() {
        return grantee;
    }


        /**
     * The simplified name of the grantee.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("granteeName")
    private final String granteeName;

        /**
     * The simplified name of the grantee.
     * @return the value
     **/
    
    public String getGranteeName() {
        return granteeName;
    }

    
        /**
     * The type of grantee.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("granteeType")
    private final GranteeType granteeType;

        /**
     * The type of grantee.
     * @return the value
     **/
    
    public GranteeType getGranteeType() {
        return granteeType;
    }

    
        /**
     * The selected permissions for a view.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("granteePermissions")
    private final java.util.List<ViewPrivilege> granteePermissions;

        /**
     * The selected permissions for a view.
     * @return the value
     **/
    
    public java.util.List<ViewPrivilege> getGranteePermissions() {
        return granteePermissions;
    }


        /**
     * The list of the columns included for permission assignment.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("columns")
    private final java.util.List<String> columns;

        /**
     * The list of the columns included for permission assignment.
     * @return the value
     **/
    
    public java.util.List<String> getColumns() {
        return columns;
    }


        /**
     * The list of the columns excluded from permission assignment.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("excludedColumns")
    private final java.util.List<String> excludedColumns;

        /**
     * The list of the columns excluded from permission assignment.
     * @return the value
     **/
    
    public java.util.List<String> getExcludedColumns() {
        return excludedColumns;
    }


        /**
     * The permission listed is inherited or not from object up in hierarchy.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isInherited")
    private final Boolean isInherited;

        /**
     * The permission listed is inherited or not from object up in hierarchy.
     * @return the value
     **/
    
    public Boolean getIsInherited() {
        return isInherited;
    }


        /**
     * Name of the object to which this permission belong to. This would be the name of view if permission is not inherited or name of object up in hierarchy if permission is inherited.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("resourceName")
    private final String resourceName;

        /**
     * Name of the object to which this permission belong to. This would be the name of view if permission is not inherited or name of object up in hierarchy if permission is inherited.
     * @return the value
     **/
    
    public String getResourceName() {
        return resourceName;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("ViewPermissionSummary(");
        sb.append("grantee=").append(String.valueOf(this.grantee));
        sb.append(", granteeName=").append(String.valueOf(this.granteeName));
        sb.append(", granteeType=").append(String.valueOf(this.granteeType));
        sb.append(", granteePermissions=").append(String.valueOf(this.granteePermissions));
        sb.append(", columns=").append(String.valueOf(this.columns));
        sb.append(", excludedColumns=").append(String.valueOf(this.excludedColumns));
        sb.append(", isInherited=").append(String.valueOf(this.isInherited));
        sb.append(", resourceName=").append(String.valueOf(this.resourceName));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof ViewPermissionSummary)) {
            return false;
        }

        ViewPermissionSummary other = (ViewPermissionSummary) o;
        return java.util.Objects.equals(this.grantee, other.grantee) &&
            java.util.Objects.equals(this.granteeName, other.granteeName) &&
            java.util.Objects.equals(this.granteeType, other.granteeType) &&
            java.util.Objects.equals(this.granteePermissions, other.granteePermissions) &&
            java.util.Objects.equals(this.columns, other.columns) &&
            java.util.Objects.equals(this.excludedColumns, other.excludedColumns) &&
            java.util.Objects.equals(this.isInherited, other.isInherited) &&
            java.util.Objects.equals(this.resourceName, other.resourceName);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.grantee == null ? 43 : this.grantee.hashCode());
        result = (result * PRIME) + (this.granteeName == null ? 43 : this.granteeName.hashCode());
        result = (result * PRIME) + (this.granteeType == null ? 43 : this.granteeType.hashCode());
        result = (result * PRIME) + (this.granteePermissions == null ? 43 : this.granteePermissions.hashCode());
        result = (result * PRIME) + (this.columns == null ? 43 : this.columns.hashCode());
        result = (result * PRIME) + (this.excludedColumns == null ? 43 : this.excludedColumns.hashCode());
        result = (result * PRIME) + (this.isInherited == null ? 43 : this.isInherited.hashCode());
        result = (result * PRIME) + (this.resourceName == null ? 43 : this.resourceName.hashCode());
        return result;
    }


}
